department-of-veterans-affairs/caseflow
# Ryan/appeals 29418

Resolves [Tasks Not Related to an Appeal - Autotexts](https://jira.devops.va.gov/browse/APPEALS-29418)

# Description
Added the autotext feature so that users can now utilize the add autotext link and generate the task instructions based off of checkboxes. 

## Acceptance Criteria

## Testing Plan

1. Navigate to http://localhost:3000/queue/correspondence/fe9173f0-38d2-42c9-af9e-2857375d542f/intake
2. Click the +add task button
3. Click add autotext
4. Notice a new modal appears
5. Check any checkboxes and click the Add button
6. Notice that the task instructions have autopopulated with your selected options.
7. Click the Add auto text button again
8. Don't check any boxes and hit Add
9. Notice that there was no autotext added and the instructions are cleared.
10. Click the Add auto text button again
11. Click the X or the Cancel button, notice that it closes the modal and modifies nothing. 

# Frontend
## User Facing Changes

 BEFORE|AFTER
 
<img width="410" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/2f6f27e9-e95b-4dcc-99e3-13179ccb8e66">
| 
<img width="604" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/02be925f-68f6-40ff-b2f9-a49fb8b0d0fc">



# Backend
## Database Changes
*Only for Schema Changes*

* [] Perform query profiling (eyeball Rails log, check bullet and fasterer output)

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
